tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/EncryptedEmailGet.scala

```diff
@@ -0,0 +1,42 @@
+package com.linagora.tmail.james.jmap.model
+
+import com.linagora.tmail.encrypted.EncryptedEmailFastView
+import org.apache.james.jmap.core.{AccountId, UuidState}
+import org.apache.james.jmap.mail.{Email, EmailIds, EmailNotFound, UnparsedEmailId}
+import org.apache.james.jmap.method.WithAccountId
+import org.apache.james.mailbox.MailboxSession
+import org.apache.james.mailbox.model.MessageId
+import reactor.core.scala.publisher.SMono
+
+case class EncryptedEmailGetRequest(accountId: AccountId,
+                                    ids: EmailIds) extends WithAccountId
+
+case class EncryptedEmailGetResponse(accountId: AccountId,
+                                     state: UuidState,
+                                     list: Option[Map[MessageId, EncryptedEmailFastView]],
+                                     notFound: Option[EmailNotFound])
+
+object EncryptedEmailFastViewResults {
+  def notFound(unparsedEmailId: UnparsedEmailId): EncryptedEmailFastViewResults =
+    EncryptedEmailFastViewResults(None, Some(EmailNotFound(Set(unparsedEmailId))))
+
+  def notFound(messageId: MessageId): EncryptedEmailFastViewResults =
+    EncryptedEmailFastViewResults(None, Some(EmailNotFound(Set(Email.asUnparsed(messageId).get))))
+
+  def list(messageId: MessageId, fastView: EncryptedEmailFastView): EncryptedEmailFastViewResults =
+    EncryptedEmailFastViewResults(Some(Map(messageId -> fastView)), None)
+
+  def empty(): EncryptedEmailFastViewResults =
+    EncryptedEmailFastViewResults(None, None)
+
+  def merge(result1: EncryptedEmailFastViewResults, result2: EncryptedEmailFastViewResults): EncryptedEmailFastViewResults = ???
+
+}
+
+case class EncryptedEmailFastViewResults(list: Option[Map[MessageId, EncryptedEmailFastView]],
+                                         notFound: Option[EmailNotFound])
+
+trait EncryptedEmailViewReader[+EncryptedEmailView] {
```

Comment:
IMO we do not even need this trait here, we can just keep the implementation (the abstraction do not add value but makes things more complex.)